service/scripts/dbSetup.ts

```diff
@@ -49,6 +49,53 @@ async function loadBundle(filePath: string) {
       let resourcesUploaded = 0;
       let notUploaded = 0;
       const uploads = bundle.entry.map(async res => {
+        // Get the main library from the Measure and add the isOwned extension on that library's
+        // entry in the relatedArtifacts of the measure
+        if (res?.resource?.resourceType && res?.resource?.resourceType === 'Measure' && res?.resource?.library) {
+          // get the main library of the measure from the library property and the version
+          const mainLibrary = res.resource.library?.[0];
+          const mainLibraryVersion = res.resource.version;
+
+          // append the version to the end of the library
+          const mainLibraryUrl = mainLibraryVersion ? mainLibrary.concat('|', mainLibraryVersion) : mainLibrary;
+
+          // check if relatedArtifacts property exists on the measure, add it if it doesn't
+          if (res.resource.relatedArtifact === undefined) {
+            res.resource.relatedArtifact = [];
+          }
+
+          // check if the main library already exists in the relatedArtifacts
+          const mainLibraryRA = res.resource.relatedArtifact.find(
+            ra => (ra.url === mainLibraryUrl || ra.resource === mainLibraryUrl) && ra.type === 'composed-of'
+          );
+
+          if (mainLibraryRA) {
+            // check if the main library's extension array exists and create it if it doesn't
+            if (mainLibraryRA.extension) {
+              // if it does exist, check that the isOwned extension is not already on it, add it if not
+              if (
+                mainLibraryRA.extension.find(
+                  e => e.url === 'http://hl7.org/fhir/StructureDefinition/artifact-isOwned' && e.valueBoolean === true
+                ) === undefined
+              )
```

Comment:
It might make more sense here to not include the `e.valueBoolean === true` case in the `.find`. I'm thinking of an edge case where an artifact has specifically been marked to *not* be an owned artifact, meaning that (1) the extension is present, and (2) `valueBoolean` is explicitly set to `false`. If this were the case, would we then have two extension entries, one with `valueBoolean` set to `true` and `valueBoolean` set to `false`?